phac-nml/irida-next
# Update dependency azure-storage-blob to azure-blob

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Fixes: #841

Azure-Storage-Blob has been deprecated in Rails 8 https://github.com/testdouble/azure-blob?tab=readme-ov-file#migration

Switching the dependency to the community gem https://github.com/testdouble/azure-blob?tab=readme-ov-file#migration

Removes api version patch from azure-storage-blob
~~Move chunk size patch from azure-storage-blob to equivalent azure-blob file~~


Related PR's:
https://github.com/phac-nml/irida-next/pull/450
https://github.com/phac-nml/irida-next/pull/422

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Install the Azurite extension for vs code https://marketplace.visualstudio.com/items?itemName=Azurite.azurite
2. install .net  `sudo apt install dotnet8`
3. install Azure Storage Explorer https://azure.microsoft.com/en-us/products/storage/storage-explorer#Download-4
4. start azurite via visual studio code commands `Azurite: Start`
5. Connect to azurite in Storage Explorer and add container
  a. Right click `Storage Accounts`, select `Connect to Azure Storage...`, select `Local storage emulator`, continue with defaults.
![image](https://github.com/user-attachments/assets/d5215736-e1f5-432f-9a7f-4d2220efbb0b)
  b. On the added local account, right click `Blob Containers` and click `Create Blob Container`, naming it `irida-next`
  c. To fix CORS errors, right click on `Blob Containers` and click `Configure CORS Settings...`. Then add a new rule and put `*` in every field.
6. in `config/environments/development.rb`, edit storage service to `config.active_storage.service = :azurite`
8. clean and build new irida db's `./scripts/clean_dev.sh`
9. You should now be able to see blobs in the Azure Storage Explorer in your irida-next
10. run irida next and verify it is working as expected, specifically file uploads and downloads.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
